hawkular-datamining-forecast/src/main/java/org/hawkular/datamining/forecast/AutomaticForecaster.java

```diff
@@ -193,4 +219,103 @@ private TimeSeriesModel selectBestModel(List<DataPoint> dataPoints) {
 
         return dataPoints;
     }
+
+    /**
+     * Strategy used for dealing with concept drift (statistical properties of modelled time series change over time)
+     * For instance originally monotonic time series changed to trend stationary or seasonal pattern showed up.
+     */
+    public abstract static class ConceptDriftStrategy {
+
+        protected AutomaticForecaster forecaster;
+
+        /**
+         * @param learnSize size of dataSet from learn method
+         * @return true if model should be selected - concept drift is present
+         */
+        public abstract boolean recalculateModel(int learnSize);
```

Comment:
the name isn't the best, when reading the code I think this reclalculate the model, because it's called recalculateModel :) What about st. like shouldRecalculateModel or shouldBeRecalculated or similar?